plugins/modules/route53.py

```diff
@@ -806,15 +816,20 @@ def main():
     rr_sets = [camel_dict_to_snake_dict(resource_record_set)]
     formatted_aws = format_record(aws_record, zone_in, zone_id)
     formatted_record = format_record(resource_record_set, zone_in, zone_id)
-
+    if module._diff:
+        module.exit_json(
+            changed=True,
+            wait_id=wait_id,
+            diff=dict(
+                before=formatted_aws,
+                after=camel_dict_to_snake_dict(formatted_record) if command_in != "delete" else {},
+                resource_record_sets=rr_sets,
+            ),
+        )
     module.exit_json(
         changed=True,
         wait_id=wait_id,
-        diff=dict(
-            before=formatted_aws,
-            after=formatted_record if command_in != "delete" else {},
-            resource_record_sets=rr_sets,
-        ),
+        resource_record_sets=[camel_dict_to_snake_dict(formatted_record)] if command_in != "delete" else {},
```

Comment:
the returned value for `resource_record_sets` should be the same whether we are running in diff mode or not